packages/opal-server/opal_server/statistics.py

```diff
@@ -58,26 +73,38 @@ def __init__(self, endpoint):
         self._endpoint: PubSubEndpoint = endpoint
         self._uptime = datetime.utcnow()
 
+        # helps us realize when another server already responded to a sync request
+        self._worker_id = uuid4().hex
+
         # state: Dict[str, List[ChannelStats]]
         # The state is built in this way so it will be easy to understand how much OPAL clients (vs. rpc clients)
         # you have connected to your OPAL server and to help merge client lists between servers.
         # The state is keyed by unique client id (A unique id that each opal client can set in env var `OPAL_CLIENT_STAT_ID`)
-        self._state: ServerStats = ServerStats(uptime=self._uptime, clients={})
+        self._state: ServerStats = ServerStats(
+            uptime=self._uptime, clients={}, servers={self._worker_id}
+        )
 
         # rpc_id_to_client_id:
         # dict to help us get client id without another loop
         self._rpc_id_to_client_id: Dict[str, str] = {}
         self._lock = asyncio.Lock()
-
-        # helps us realize when another server already responded to a sync request
-        self._worker_id = uuid4().hex
         self._synced_after_wakeup = asyncio.Event()
         self._received_sync_messages: Set[str] = set()
+        self._publish_tasks = TasksPool()
```

Comment:
All running tasks should be awaited / cancelled on shutdown to gracefully terminate.
Could use [Python's asyncio.TaskGroup](https://docs.python.org/3/library/asyncio-task.html#task-groups) instead.

Reply:
I think `asyncio.TaskGroup` was only introduced in 3.11 and OPAL currently supports 3.9 & 3.10.
`TasksPool` adds a done callback that awaits each task when its done